synapse/models/inet.py

```diff
@@ -149,20 +164,90 @@ def getDataModel():
             ('inet:netuser',{'ptype':'inet:netuser'},[
                 ('site',{'ptype':'inet:fqdn','ro':1}),
                 ('user',{'ptype':'inet:user','ro':1}),
-                ('name',{'ptype':'str','defval':'??'}),
+
+                ('dob',{'ptype':'time'}),
+
+                #('bio:bt',{'ptype':'wtf','doc':'The netusers self documented blood type'}),
+
+                ('url',{'ptype':'inet:url'}),
+                ('webpage',{'ptype':'inet:url'}),
+                ('avatar',{'ptype':'file:bytes'}),
+
+                ('tagline',{'ptype':'str:txt','doc':'A netuser status/tag line text'}),
+                ('occupation',{'ptype':'str:txt','doc':'A netuser self declared occupation'}),
+                #('gender',{'ptype':'inet:fqdn','ro':1}),
+
+                ('name',{'ptype':'inet:user'}),
+                ('realname',{'ptype':'ps:name'}),
                 ('email',{'ptype':'inet:email'}),
+                ('phone',{'ptype':'tel:phone'}),
                 ('signup',{'ptype':'time','defval':0,'doc':'The time the netuser account was registered'}),
+                ('signup:ipv4',{'ptype':'inet:ipv4','defval':0,'doc':'The original ipv4 address used to sign up for the account'}),
                 ('passwd',{'ptype':'inet:passwd','doc':'The current passwd for the netuser account'}),
                 ('seen:min',{'ptype':'time:min'}),
                 ('seen:max',{'ptype':'time:max'}),
             ]),
 
-            ('inet:netmesg',{'ptype':'inet:netmesg'},[
+            ('inet:netgroup',{},[
                 ('site',{'ptype':'inet:fqdn','ro':1}),
-                ('to',{'ptype':'inet:user','ro':1}),
-                ('from',{'ptype':'inet:user','ro':1}),
-                ('sent',{'ptype':'time','ro':1,'doc':'The time at which the message was sent'}),
-                ('body',{'ptype':'str'}),
+                ('name',{'ptype':'ou:name','ro':1}),
+
+                ('desc',{'ptype':'str:txt'}),
+
+                ('url',{'ptype':'inet:url'}),
+                ('webpage',{'ptype':'inet:url'}),
+                ('avatar',{'ptype':'file:bytes'}),
+            ]),
+
+            ('inet:netpost',{},[
+
+                ('netuser',{'ptype':'inet:netuser','ro':1}),
+
+                ('netuser:site',{'ptype':'inet:fqdn','ro':1}),
+                ('netuser:user',{'ptype':'inet:user','ro':1}),
+
+                ('time',{'ptype':'time','ro':1}),
+
+                ('replyto',{'ptype':'inet:netpost'}),
+
+                #FIXME how do we deal with this issue both generally and in a non-english-centric way
+                ('text',{'ptype':'str:txt','defval':'??','doc':'The text of the post'}),
+                ('text:en',{'ptype':'str:txt','doc':'An optional english translation of the post text'}),
+
+                ('url',{'ptype':'inet:url','doc':'The (optional) URL where the post is published/visible'}),
+                ('file',{'ptype':'file:bytes','doc':'The (optional) file which was posted'}),
```

Comment:
Netpost content may include links that are of interest - e.g., a link to an exploit or malicious document. Should embedded links be called out as secondary props, or would they somehow be "parseable" from the netpost text for searching / pivoting?